src/s_sdl_paint_renderer.h

```diff
@@ -1,37 +1,39 @@
 
-#ifndef _h_RetcangleRenderer
-#define _h_RectangleRenderer
+#ifndef _h_SDLPaintRenderer
+#define _h_SDLPaintRenderer
 
 #include <cstdint>
 
 #include "lib/entity.h"
-#include "lib/renderer.h"
+#include "lib/sdl_renderer.h"
 #include "lib/system.h"
 
 namespace spacegun {
   using std::string;
   using aronnax::Color;
   using aronnax::Entity;
   using aronnax::Entities;
+  using aronnax::SDLRenderer;
   using aronnax::System;
 
-  class RectangleRenderer: public System
+  class SDLPaintRenderer: public System
   {
     public:
-      RectangleRenderer();
-      RectangleRenderer(IRenderer* renderer);
+      SDLPaintRenderer();
+      SDLPaintRenderer(SDLRenderer* renderer);
```

Comment:
This would be a good place to try a class template.

template <class TRenderer>

new PaintRenderer<SDLRenderer>();
new PaintRenderer<GPURenderer>();